chain/chain/src/chain.rs

```diff
@@ -4436,16 +4441,41 @@ impl<'a> ChainUpdate<'a> {
                 me,
                 block,
                 &prev_block,
-                &receipts_by_shard,
+                &incoming_receipts,
                 ApplyChunksMode::NotCaughtUp,
             )?
         };
 
+        Ok((apply_chunk_work, BlockPreprocessInfo { state_dl_info, incoming_receipts }))
+    }
+
+    /// Runs the block processing, including validation and finding a place for the new block in the chain.
+    /// Returns new head if chain head updated, as well as a boolean indicating if we need to start
+    ///    fetching state for the next epoch.
+    fn process_block(
+        &mut self,
+        me: &Option<AccountId>,
+        block: &MaybeValidated<Block>,
+        provenance: &Provenance,
+        on_challenge: &mut dyn FnMut(ChallengeBody),
+    ) -> Result<Option<Tip>, Error> {
+        let _span =
+            tracing::debug_span!(target: "chain", "Process block", "#{}", block.header().height())
+                .entered();
+
+        let (apply_chunk_work, BlockPreprocessInfo { state_dl_info, incoming_receipts }) =
```

Comment:
I'd add a comment - that here we do all the processing that doesn't require any changes to the storage.

Reply:
I'll add more comments in the later PRs because I may still move the code around later.